### PR TITLE
add flag --no-push

### PR DESCRIPTION
**Pull request by raunyoliveira-hotmart**
Add flag `--no-push`, reference: [https://github.com/GoogleContainerTools/kaniko#--no-push](https://github.com/GoogleContainerTools/kaniko#--no-push)

Use case:

We want to build our images without pushing it when someone opens a pull request in our repository.

Example:

```
/kaniko/executor \
--dockerfile=./xvfb/Dockerfile \
--context=dir://. \
--destination=public.ecr.aws/[redacted] \
--cache=true \
--cache-repo=public.ecr.aws/[redacted] --no-push

INFO[0019] Skipping push to container registry due to --no-push flag 
```

image

@shubham149 can you review it please?

### CODE REVIEW

File: cmd/kaniko-docker/main.go [hunk 1]
```diff
@@ -119,6 +119,11 @@ func main() {
 			Usage:  "Cache timeout in hours. Defaults to two weeks.",
 			EnvVar: "PLUGIN_CACHE_TTL",
 		},
+		cli.StringFlag{
```
Comment (shubham149) [inline cmd/kaniko-docker/main.go:122]: use a boolean flag instead of string flag

Comment (raunyoliveira-hotmart) [reply cmd/kaniko-docker/main.go]: oh my bad! done!

File: kaniko.go [hunk 2]
```diff
@@ -82,6 +83,10 @@ func (p Plugin) Exec() error {
 		cmdArgs = append(cmdArgs, fmt.Sprintf("--cache-ttl=%d", p.Build.CacheTTL))
 	}
 
+	if p.Build.NoPush == true {
```
Comment (shubham149) [inline kaniko.go:86]: Minor change: 

`if p.Build.NoPush {
  ...
}`

Comment (raunyoliveira-hotmart) [reply kaniko.go]: done.

**Pull request merged**